app/src/controllers/reservation/services/ReservationService.ts

```diff
@@ -167,6 +213,30 @@ const ReservationService: ReservationServiceInterface = {
       console.error('Invalid date, earlier than today')
       throw new InvalidParamsError()
     }
+    const filteredDates = filterAndGetWantedDate(reservationDate)
+
+    const reservationDatesAndDuration = await ReservationRepository.fetchReservationsByDSS(
+      shopId, filteredDates.startDate,
+      filteredDates.endDate, stylistId,
+    )
+
+    const shopSchedule = await ShopRepository.fetchShopSchedule(shopId)
+
+    const menuDuration = await ReservationRepository.fetchMenuDuration(menuId)
+
+    const checkWithShopSchedule = checkInsideShopSchedule(shopSchedule, reservationDate, menuDuration)
+
+    if (!checkWithShopSchedule) {
+      throw new OutOfScheduleError()
+    }
+    if (menuDuration === 0) {
+      throw new InvalidParamsError()
+    }
+    const availability = checkAvailablity(reservationDatesAndDuration, reservationDate, menuDuration)
```

Comment:
"isAvailable" is more informative than "availability", it's a noun, it does not describe anything